src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,205 @@
+        } catch (SQLException e) {
+            throw new RuntimeException("Problem updating info for car: " + car + ".", e);
+        }
+        removeDrivers(car.getId());
+        saveDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Problem deleting car by id: " + id + ".", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, model, m.name, country, m.id AS manufacturer_id "
+                + "FROM cars_drivers cd "
+                + "JOIN cars c "
+                + "ON cd.car_id = c.id "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = 0;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(composeCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Problem getting cars from DB by driver id: "
+                    + driverId + ".", e);
+        }
+        if (!cars.isEmpty()) {
+            cars.forEach(c -> c.setDrivers(getDrivers(c.getId())));
+        }
+        return cars;
+    }
+
+    private void saveDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Problem saving drivers list to db for car: " + car + ".", e);
+        }
+    }
+
+    private Car composeCar(ResultSet resultSet) throws SQLException {
+        return new Car(resultSet.getObject("id", Long.class),
+                resultSet.getString("model"),
+                new Manufacturer(resultSet.getObject("manufacturer_id", Long.class),
+                        resultSet.getString("name"), resultSet.getString("country")));
```

Comment:
it's hard to read, let's use setters or create variables for storing values from resultSet to pass them in constructors

Reply:
Done